utils/stdx/src/lib.rs

```diff
@@ -98,31 +102,31 @@ pub fn as_chunks<const N: usize, T>(slice: &[T]) -> (&[[T; N]], &[T]) {
     (head, tail)
 }
 
-/// Like `as_chunks` but returns an error if there’s a remainder.
-pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Result<&[[T; N]], ()> {
+/// Chunks if `as_chunks` has no remainder.
+pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Option<&[[T; N]]> {
```

Comment:
I think we should keep using a `Result` here. It fits better semantically in that a failure here signals that the call is ill formed.

An error type like 

```rust
#[derive(Debug, Clone)]
struct InexactChunkingError(());
impl std::error::Error for InexactChunkingError {}
impl std::fmt::Display for InexactChunkingError { ... }
```

probably isn’t too onerous to add for this.

Reply:
makes sense, thanks for the suggestion